### PR TITLE
Adjust icon set to integrate client icons

### PR DESCRIPTION
**Pull request by lyzadanger**
It turns out what I believed was an exercise in figuring out how to generate icons in other applications, as captured in https://github.com/hypothesis/frontend-shared/issues/588, was actually a bookkeeping and consolidation exercise.

The icons that the client registers and has source for locally are icons that exist (or partially exist, or really should exist) in the shared package, but are _different_. Some of the icons shared-package icon set are/were aspirational or proposed, while the client is still using icons designed before that design project. e.g. the "Reply" icon in the shared package is (well, was, after these changes) a newer icon that more closely matches the style of other newer icons, but the client uses a different "Reply" icon.

So what I've done is make the shared package reflect reality, while still retaining the aspirational icons in using `*Alt` naming conventions. This doesn't leave any icons in the client that need to be generated. We may well need custom app icons in the future, but let's cross that bridge when we get there.

**After these changes, the shared package icons correctly represent the full set of known application icons.**[^1]

Exhaustively, these changes were:

* Added `MenuExpand` and `MenuCollapse`. `MenuExpand` is used by the client in menu-disclosure interfaces[^2]. The client currently uses a local CSS transform to achieve what `MenuCollapse` looks like (i.e. `MenuCollapse` is "new").
* Update `Lock`, `Reply` and `Sort` icons to reflect what is in current use in the client
* Add `LockAlt` (and `LockAltFilled`), `ReplyAlt` and `SortAlt` to hold alternate variants of those icons (proposed/aspirational)
* Added `PointerUp` and `PointerDown`. These are based on the `pointer` icon from the client. The idea is that these get positioned at the edge of pop-up interfaces (e.g. "Share this annotation" panel) and "point back" at the button or interface element that spawned them. I had to do some rework on this icon, because it didn't actually fit in its viewbox and was off-center (weird). If you were to look at the old version closely, it's actually "snub" because the tip of the point is outside of the viewbox. Similar to `MenuExpand`, the client currently uses CSS transforms to rotate a single `pointer` icon, but I've added down and up variants here for convenience.

Fixes https://github.com/hypothesis/frontend-shared/issues/588

[^1]: LMS does not have any local icons
[^2]: LMS inconsistently uses `CaretDown` (versus `MenuExpand`) for a menu-disclosure interface in the `StudentSelector`; I'll address this as we update components in that app.

### CODE REVIEW

File: src/icons.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ export { default as listOrdered } from '../images/icons/list-ordered.svg';
 export { default as listUnordered } from '../images/icons/list-unordered.svg';
 export { default as list } from '../images/icons/list.svg';
 export { default as lock } from '../images/icons/lock.svg';
-export { default as lockFilled } from '../images/icons/lock-filled.svg';
+export { default as lockFilled } from '../images/icons/lock-alt-filled.svg';
```
Comment (lyzadanger) [inline src/icons.js:68]: This module registers icons for the "legacy" coverage in the pattern library. I've left the export name as `lockFilled` so I don't need to go in and update legacy pattern-library pages.

**Review by robertknight — APPROVED** [review]
LGTM. Is there a plan to change the client to either use the "Alt" icons, or retire them if it is decided to keep the existing icon? Having both may create some confusion over which should be used in future projects.

File: src/components/icons/Sort.js [hunk 2]
```diff
@@ -13,12 +13,19 @@ export default function SortIcon(props) {
       height="16"
       viewBox="0 0 16 16"
       aria-hidden="true"
+      class="Icon Icon--sort"
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply src/components/icons/Sort.js:16]: No! Nice catch.

File: src/components/icons/PointerUp.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+/**
+ * Icon generated from pointer-up.svg
+ *
+ * @param {import('preact').JSX.SVGAttributes<SVGSVGElement>} props
+ */
+export default function PointerUpIcon(props) {
+  return (
+    <svg width="16" height="9" xmlns="http://www.w3.org/2000/svg" {...props}>
+      <path d="m.5 9 7-8 8 8" stroke="currentColor" class="layer" />
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply src/components/icons/PointerUp.js:11]: Yes. I can remove it from source, though, and regenerate, no problem.

**Pull request merged**